### ISSUE TITLE
Not closing in long positions after short position and exit_signal

### ISSUE THREAD

**Issue by bit-storm** [freqtrade/freqtrade#8531]
## Describe my environment

  * Operating system: linux ubuntu 20
  * Python Version:  3.8
  * CCXT version:  2.5.12
  * Freqtrade Version: 2022.12

  
This is my setup for the strategy:

'''
# ROI table:
    minimal_roi = {
        "0": 0.99,
    }

    # Stoploss:
    stoploss = -0.99

    timeframe = '5m'

    can_short = True
    
    startup_candle_count = 600
    process_only_new_candles = False

    use_exit_signal = True
    exit_profit_only = False
    ignore_roi_if_entry_signal = True
    
'''

Hello, thanks for your good and useful library.
I have a strategy that works in five minutes.
I have no stop loss and roi and both are set at 99%.
I use exit_signal for all exits.
My problem is that in the live robot test, many of these exit signals do not leave the strategy.
Also, if you pay attention to all the entry and exit points of the photo, you will notice that the exit is not considered in the short positions, but the entry points to the long are also the exit points.

![Screenshot from 2023-04-20 22-23-09](https://user-images.githubusercontent.com/130432149/233464987-ebcc9d83-40d9-40fa-b2bf-4442b1ee96c9.png)

Thank you for helping me solve this problem.

**Comment by xmatthias** [comment]
Well the screenhshot shows a long trade (there is only one single trade in the screenshot, the green triangles are signals only), and a few short exit signals (pins).

There's also no long exit signal in the screenshot (at least none visible) - which would be a yellow / ocker diamond.

As far as this screenshot is concerned, i'd therefore say it's acting correctly.

All symbols lined out, in the sequence of short, short exit, long entry, long exit.

![image](https://user-images.githubusercontent.com/5024695/233541803-dc94393f-daaf-4194-b997-7ad55b4605ec.png)
